### PR TITLE
Unit tests for multi triggers MC-2344

### PR DESCRIPTION
**Pull request by piyush-kukadiya**
Added unit tests for multi trigger feature https://github.com/CleverTap/clevertap-android-sdk/pull/683

1. Fixes broken unit tests.
2. Adds new test cases

### CODE REVIEW

**Review by CTLalit — COMMENTED** [review]
Can we change code design to not use so many mockStatic blocks? Not sure about the scope of change, let me know.

**Comment by piyush-kukadiya** [comment]
@CTLalit We have many static classes/functions which requires mockStatic blocks. It is how code is designed surrounding `CleverTapAPI` and `CTExecutorFactory`. We have it in our backlog to remove static usage from codebase, but it's very long process. We can pick it up as small chunks.

File: clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt [hunk 1]
```diff
@@ -605,6 +599,188 @@ class CleverTapAPITest : BaseTestCase() {
...
+        mockStatic(CTExecutorFactory::class.java).use {
+            `when`(CTExecutorFactory.executors(any())).thenReturn(
+                MockCTExecutors(
+                    cleverTapInstanceConfig
+                )
+            )
+            mockStatic(CleverTapFactory::class.java).use {
+                `when`(CleverTapFactory.getCoreState(application, cleverTapInstanceConfig, null))
+                    .thenReturn(corestate)
+                cleverTapAPI = CleverTapAPI.instanceWithConfig(application, cleverTapInstanceConfig)
```
Comment (CTLalit) [inline clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt:619]: this is common block in all test cases, can we extract?

**Review by CTLalit — APPROVED** [review]

File: clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt (hunk 1 above)
Comment (piyush-kukadiya) [reply clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt]: fixed in https://github.com/CleverTap/clevertap-android-sdk/commit/cde4f26382c4e1d147db3583d528dd34b6ac0167

File: clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt [hunk 2]
```diff
@@ -21,381 +22,299 @@ class CleverTapAPITest : BaseTestCase() {
...
-
-        mockStatic(CTExecutorFactory::class.java).use {
-            `when`(CTExecutorFactory.executors(any())).thenReturn(
-                    MockCTExecutors(
-                            cleverTapInstanceConfig
-                    )
+        setupBasicTestWithAny {
+            val bundle = Bundle()
+            val lock = Object()
+            //CleverTapAPI.getDefaultInstance(application)
```
Comment (vasct) [inline clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt:272]: Commented code

File: clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt [hunk 3]
```diff
@@ -586,23 +446,139 @@ class CleverTapAPITest : BaseTestCase() {
...
+            assertEquals(1, userEventLogCountActual)
+            verify(corestate.localDataStore).readUserEventLogCount(evt)
+        }
+    }
+
+    @Test
+    fun `test getUserEventLog`(){
+        // Arrange
+        val evt = "test"
+        val log = UserEventLog(evt,1000L,1000L,1,"dId")
```
Comment (vasct) [inline clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt:481]: There are already a lot of changes to this file, you can format the whole thing.

**Review by vasct — APPROVED** [review]

File: clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt [hunk 4]
```diff
@@ -21,381 +22,299 @@ class CleverTapAPITest : BaseTestCase() {
...
+                    ArgumentMatchers.any(),
+                    ArgumentMatchers.any()
+                )
+            )
+                .thenReturn(corestate)
+            block()
+        }
+    }
+
+    private fun setupBasicTest(block: () -> Unit) {
```
[the remark this answers is not in the document]
Comment (piyush-kukadiya) [reply clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt:57]: fixed in https://github.com/CleverTap/clevertap-android-sdk/commit/5a2f2c022aac769fc693f5ff6400bc6bfde61628

File: clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt (hunk 2 above)
Comment (piyush-kukadiya) [reply clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt:272]: fixed in https://github.com/CleverTap/clevertap-android-sdk/commit/5a2f2c022aac769fc693f5ff6400bc6bfde61628

File: clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt (hunk 3 above)
Comment (piyush-kukadiya) [reply clevertap-core/src/test/java/com/clevertap/android/sdk/CleverTapAPITest.kt:481]: fixed in https://github.com/CleverTap/clevertap-android-sdk/commit/5a2f2c022aac769fc693f5ff6400bc6bfde61628

**Pull request merged**